### ISSUE TITLE
Add Safari Support 

### ISSUE THREAD

**Issue by unbeatable-101** [Sainan/Universal-Bypass#1429]
In Safari 14 the webextension API will be added, will the extension be able to be added?

**Comment by Sainan** [comment]
That's a question I can't answer, but I'll look into it when the time comes.

**Comment by unbeatable-101** [comment]
It might not, the API is not completely supported.

**Comment by Sainan** [comment]
I'm not a time traveller, I don't have secret insights at Apple, so I can't tell you what will happen.

**Comment by LoYungSum** [comment]
Big Sur public beta is released, just wanna know did anyone check if it's compatible with Safari?

**Comment by FireMasterK** [comment]
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Browser_support_for_JavaScript_APIs

We need to see if Universal-Bypass can work with these missing functionality.

If it can't you'll have to wait for Apple.

**Comment by Sainan** [comment]
Well, Universal Bypass is a WebExtension, so either the Chrome or Firefox build can be loaded into Safari in developer mode maybe?

**Comment by FireMasterK** [comment]
Perhaps, I don't use any Apple devices, so maybe someone else over here can report their findings on what's broken and what's not.

Also, it seems like Apple's going to charge their annual fee of 100$ a year if you want to put it on the App Store. Here's a quote from their website:

"Not all Chrome, Firefox, and Edge extensions will work in Safari, and developers will be required to pay $99 per year for an Apple Developer Program membership in order to port their extensions to Safari. Apple has a WWDC video and documentation with more details for developers."


The process for developing/installing the extension seems tedious without paying Apple's tax: https://developer.apple.com/documentation/safariservices/safari_app_extensions/building_a_safari_app_extension#2957926

You might need xcode if you even if you just want to install it.

The worst part about the entire thing: you may need to write some things in swift/c if you want it to even function, here's a quote from their documentation:
"New Xcode projects require you to select either the Swift or Objective-C language for native development. For your Safari Web Extension, you may not need to make any native customizations at all because your extension uses the JavaScript, HTML, and CSS you provide. If you are unsure which language to use, select Swift."

Goodluck to anyone who tries to port WebExtensions to Safari 😄

TLDR: This is more suitable for a port, and is definitely not worth the effort required, it's definitely a lot more complex than a developer mode setting.

**Comment by Sainan** [comment]
If it does not support the extension as a WebExtension, I won't bother porting it. I'm also not gonna reward Apple with $99/year for being a shitty company just to have my free extension available to their users.
